### PR TITLE
[tests-only] [full-ci] test for occ versions command

### PR DESCRIPTION
**Pull request by sakshamgurung**
## Description
- Adds tests of [occ versions command](https://doc.owncloud.com/server/admin_manual/configuration/server/occ_command.html#versions)
- [x] `sudo -u www-data php occ versions:cleanup [<user_id>]...`
- [ ] `sudo -u www-data php occ versions:expire [<user_id>]...`
## Related Issue

- Working on issue: https://github.com/owncloud/core/issues/39165
- Working on issue list: https://github.com/owncloud/core/issues/33051

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by sakshamgurung** [comment]
Having problem with turning off auto handling of expired versions of files. Tried setting `  'trashbin_retention_obligation' => 'disabled'` in config.php but still shows following cli message: `Auto expiration is configured - expiration will be handled automatically.` when running occ cmd: `sudo -u www-data php occ versions:expire [<user_id>]...`

File: tests/acceptance/features/cliMain/fileVersions.feature [hunk 1]
```diff
@@ -11,3 +11,31 @@ Feature: check file versions
...
+    And user "Brian" has uploaded file with content "version 2" to "/sharingfolder/sharefile.txt"
+    And user "Brian" has uploaded file with content "version 3" to "/sharingfolder/sharefile.txt"
+    Then the version folder of file "sharingfolder/sharefile.txt" for user "Alice" should contain "3" element
+    And the version folder of file "sharingfolder/sharefile.txt" for user "Brian" should contain "3" element
+    When the administrator deletes all the versions for following users:
+      | username |
+      | Alice    |
+      | Brian    |
+    Then the version folder of file "/sharingfolder/sharefile.txt" for user "Alice" should contain "0" element
+    Then the version folder of file "/sharingfolder/sharefile.txt" for user "Brian" should contain "0" element
```
Comment (kiranparajuli589) [inline tests/acceptance/features/cliMain/fileVersions.feature:36]:
```suggestion
    And the version folder of file "/sharingfolder/sharefile.txt" for user "Brian" should contain "0" element
```

File: tests/acceptance/features/cliMain/fileVersions.feature [hunk 2]
```diff
@@ -11,3 +11,31 @@ Feature: check file versions
...
+    And user "Alice" has uploaded file with content "version 1" to "/sharingfolder/sharefile.txt"
+    And user "Alice" has uploaded file with content "version 2" to "/sharingfolder/sharefile.txt"
+    And user "Alice" has uploaded file with content "version 3" to "/sharingfolder/sharefile.txt"
+    And user "Brian" has created folder "/sharingfolder"
+    And user "Brian" has uploaded file with content "old content" to "/sharingfolder/sharefile.txt"
+    And user "Brian" has uploaded file with content "version 1" to "/sharingfolder/sharefile.txt"
+    And user "Brian" has uploaded file with content "version 2" to "/sharingfolder/sharefile.txt"
+    And user "Brian" has uploaded file with content "version 3" to "/sharingfolder/sharefile.txt"
+    Then the version folder of file "sharingfolder/sharefile.txt" for user "Alice" should contain "3" element
+    And the version folder of file "sharingfolder/sharefile.txt" for user "Brian" should contain "3" element
```
Comment (kiranparajuli589) [inline tests/acceptance/features/cliMain/fileVersions.feature:30]: `then` step for `setup` steps is not required.
the expectations are/(should be) asserted inside those `given` steps

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 3]
```diff
@@ -2438,18 +2462,52 @@ public function theAdministratorEmptiesTheTrashbinOfUserUsingTheOccCommand($user
...
+	 *
+	 * @param string $user
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function theAdministratorDeletesAllTheExpiredVersionsForUser(string $user):void {
+		$user = $this->featureContext->getActualUsername($user);
+		$this->deleteAllExpiredVersionsForUserUsingOccCommand($user);
+	}
```
Comment (kiranparajuli589) [inline tests/acceptance/features/bootstrap/OccContext.php:2509]: please do not use the same implementations for `given` and `when` steps.

- for the `given` step you should check that all the expired versions for the user are actually deleted
- for the `when` step you just perform the action and then assert the expectation on `then` steps

**Comment by phil-davis** [comment]
>  Having problem with turning off auto handling of expired versions of files. 

https://github.com/owncloud/core/blob/master/apps/files_trashbin/lib/Command/ExpireTrash.php has code:

```
	protected function execute(InputInterface $input, OutputInterface $output) {
		$retentionEnabled = $this->trashExpiryManager->retentionEnabled();
		if (!$retentionEnabled) {
			$output->writeln("Auto expiration is configured - expiration will be handled automatically.");
			return 1;
		}

...
```

It only emits one message. If the retention obligation is set to "auto" or "disabled" then it always emits the same message. That needs to be fixed in the code.

**Comment by phil-davis** [comment]
https://github.com/owncloud/core/blob/master/apps/files_trashbin/lib/TrashExpiryManager.php has methods
`expiryEnabled()` - tells if `retentionObligation` is not "disabled". If this is `false` then we will know to write a message that trashbin expiration is disabled.

**Review by dpakach — COMMENTED** [review]
lgtm 👍🏼 but the CI is failing in some unrelated test please investigate that.

**Comment by phil-davis** [comment]
Probably just rebase and force-push and see what happens to CI.

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm 👍🏼

**Review by phil-davis — APPROVED** [review]

**Pull request merged**